### DISCUSSION TITLE
RFC: Extending SBLT system / Semi unification with SBLT

### DISCUSSION THREAD

**Discussion by Luffyyy** [diesel-modding/PAYDAY-2-BeardLib#683] [Ideas]
BeardLib is a library that loads its own mods. The reasons that it did so and not just use BLT directly:
- BLT doesn't support mod_overrides,
- It was convenient to have everything in one place.

BeardLIb 5.0 attempts to change the modding in PD2 to allow you to install all mods in the `mods` folder. This will allow more mods to also utilize SBLT. And so the idea of defining BeardLib modules in supermod.xml is a very natural progression of the idea.

This idea comes to redefine a major part of PD2 modding pushing people to make a SBLT mod first and only then adding BeardLib's modules to supermod.xml directly. Of course, some specific mods that heavily utilize BeardLib or add new content are not really the point of this change, but they will start to look more like SBLT mods.

Why?
- Reduce confusion and have new modders learn SBLT and then if they decide, start using BeardLib.
- Reduce complexity of BeardLib making it a more true library than a mod loader on top of an existing mod loader.

Old mods
Of course, we don't have any intention of breaking old mods and they will still work, but this change may deprecate some of these features (they will still work just not get updated anymore). Depending how well it goes.

There are some difficult parts to figure out, it's possible this will be part of BeardLib 5 and possibly not.

**Comment by segabl** [comment]
So basically, when supermod.xml supports all mod.txt features, you could add BeardLib specific things to it and BeardLib would either read the "leftover" mod data after the mod  instance has been created or re-read the xml.

**Reply by Luffyyy** [reply]
Yeah, preferably hook into somewhere and get the mod data.

**Comment by Luffyyy** [comment]
Though the more I think about it, I think it's better be an option than forced. Might cause more headaches with custom maps since BLE has to edit the main.xml and that won't bode well with SBLT's custom one (plus extra coding)

**Reply by segabl** [reply]
I think maps could have their own xml files, like they already partly do

**Reply by Luffyyy** [reply]
yeah that's a possible way to tackle that,
